src/lavinmq/mqtt/session.cr

```diff
@@ -36,15 +36,11 @@ module LavinMQ
             consumer.deliver(pub_packet)
           end
           Fiber.yield if (i &+= 1) % 32768 == 0
+        rescue ex
+          @log.error(exception: ex) { "Failed to deliver message in deliver_loop" }
+          @consumers.each &.close
+          client = nil
```

Comment:
I think `self` is required here, otherwise it's a local variable assignment. 
```suggestion
          self.client = nil
```

Reply:
ah good catch!